packages/core/test/insured-bridge/BridgePool.js

```diff
@@ -545,8 +563,156 @@ describe("BridgePool", () => {
+
+      // Grab OO price request information from Relay struct.
+      const relayStatus = await bridgePool.methods.relays(depositHash).call();
+
+      // Invalid instant relay with incorrect fee sends incorrect amount to user
+      const invalidRealizedLpFeeAmount = toBN(invalidRealizedLpFee)
+        .mul(toBN(relayAmount))
+        .div(toBN(toWei("1")));
+      const invalidInstantRelayAmountSubFee = toBN(relayAmount)
+        .sub(invalidRealizedLpFeeAmount)
+        .sub(realizedSlowRelayFeeAmount)
+        .sub(realizedInstantRelayFeeAmount)
+        .toString();
+      await l1Token.methods
+        .approve(bridgePool.options.address, invalidInstantRelayAmountSubFee)
+        .send({ from: instantRelayer });
+      await bridgePool.methods.speedUpRelay(depositData).send({ from: instantRelayer });
+      const startingInstantRelayerAmount = (await l1Token.methods.balanceOf(instantRelayer).call()).toString();
+
+      // User receives invalid instant relay amount.
+      assert.equal(
+        (await l1Token.methods.balanceOf(depositData.l1Recipient).call()).toString(),
+        invalidInstantRelayAmountSubFee
+      );
+
+      // Before slow relay expires, it is disputed.
+      const _relayAncillaryData = await bridgePool.methods
+        .getRelayAncillaryData(depositData, { ...relayData, realizedLpFeePct: invalidRealizedLpFee })
+        .call();
+      await l1Token.methods.approve(optimisticOracle.options.address, totalRelayBond).send({ from: disputer });
+      await optimisticOracle.methods
+        .disputePrice(
+          bridgePool.options.address,
+          defaultIdentifier,
+          relayStatus.priceRequestTime.toString(),
+          _relayAncillaryData
+        )
+        .send({ from: disputer });
+
+      // While dispute is pending resolution, a valid relay is resubmitted. Advance time so that
+      // price request data is different.
+      await l1Token.methods.mint(rando, totalRelayBond).send({ from: owner });
+      await l1Token.methods.approve(bridgePool.options.address, totalRelayBond).send({ from: rando });
+      await advanceTime(1);
+      const requestTimestamp = (await bridgePool.methods.getCurrentTime().call()).toString();
+      await bridgePool.methods.relayDeposit(...generateRelayParams()).send({ from: rando });
+
+      // Instant relayer address should be empty for most recent relay with valid params.
+      const instantRelayHash = generateInstantRelayHash(depositHash, relayData);
+      assert.equal(await bridgePool.methods.instantRelays(instantRelayHash).call(), ZERO_ADDRESS);
+
+      // Can speed up relay with new (valid) params.
+      await l1Token.methods.mint(instantRelayer, instantRelayAmountSubFee).send({ from: owner });
+      await l1Token.methods
+        .approve(bridgePool.options.address, instantRelayAmountSubFee)
+        .send({ from: instantRelayer });
+      await bridgePool.methods.speedUpRelay(depositData).send({ from: instantRelayer });
```

Comment:
this test case has the initial (invalid) instant relayer and subsequent (valid) instant relayer as the same EOA address. it might be a good idea to have a separate test case such that these are different EOAs. I'm pretty sure this will behave exactly the same but it won't hurt to validate it, especially given that is a likely flow in production.

Reply:
Yeah my test case checks that the `instantRelayer` only gets a refund for the second correct instant relay so I think this test logic won't change